src/botPage/view/blockly/utils.js

```diff
@@ -213,6 +213,16 @@ export const getTopBlocksByType = type => Blockly.mainWorkspace.getTopBlocks().f
 
 export const getMainBlocks = () => config.mainBlocks.map(blockType => getBlockByType(blockType)).filter(b => b);
 
+export const hasAllRequiredBlocks = workspace => {
+    const blocksInWorspace = workspace.getAllBlocks();
```

Comment:
There's a minor typo here.

```suggestion
    const blocksInWorkspace = workspace.getAllBlocks();
```